src/app_service/service/node_configuration/service.nim

```diff
@@ -46,20 +46,45 @@ method init*(self: Service) =
     error "error: ", errDesription
     return
 
-proc saveConfiguration(self: Service, configuration: NodeConfigDto) =
+proc saveConfiguration(self: Service, configuration: NodeConfigDto): bool =
   if(not self.settingsService.saveNodeConfiguration(configuration.toJsonNode())):
     error "error saving node configuration "
-    return
+    return false
   self.configuration = configuration
+  return true
   
 method getWakuVersion*(self: Service): int =
   if self.configuration.WakuConfig.Enabled:
     return WAKU_VERSION_1
   elif self.configuration.WakuV2Config.Enabled:
     return WAKU_VERSION_2
+  
+  error "unsupported waku version"
   return 0
 
-method setWakuVersion*(self: Service, wakuVersion: int) =
+method getBloomLevel*(self: Service): string =
+  var bloomFilterMode: bool
+  var fullNode: bool
+
+  let wakuVersion = self.getWakuVersion()
+  if wakuVersion == WAKU_VERSION_1:
+    bloomFilterMode = self.configuration.WakuConfig.BloomFilterMode
+    fullNode = self.configuration.WakuConfig.FullNode
+  elif wakuVersion == WAKU_VERSION_2:
+    bloomFilterMode = self.configuration.WakuV2Config.BloomFilterMode
+    fullNode = self.configuration.WakuV2Config.FullNode
+  else:
+    return BLOOM_LEVEL_NORMAL
+
+  if (bloomFilterMode):
+    if(fullNode):
+      return BLOOM_LEVEL_FULL
+    else:
+      return BLOOM_LEVEL_NORMAL
+  else:
+    return BLOOM_LEVEL_LIGHT
+
```

Comment:
Bloom levels only apply to WakuV1. This function should either return empty or throw an exception if called when the current waku version is 2